### DISCUSSION TITLE
Using the built-in emmet tools in helix editor

### DISCUSSION THREAD

**Discussion by overbyte** [vuejs/language-tools#4483] [General]
Hi

I'm attempting to get emmet to handle self-closing tags in .vue files.

When using the emmet-language-server I got the following result when trying `self/` and then hitting tab: 

In a JSX file: 
![jsx](https://github.com/olrtg/emmet-language-server/assets/402785/22a6ebc2-80a1-40ad-9017-1d6b7c5e9727)

In a vue file:
![vue](https://github.com/olrtg/emmet-language-server/assets/402785/ee58181c-d73b-446d-8e97-53e630493dbf)

I asked this question in the emmet-language-server (https://github.com/olrtg/emmet-language-server/issues/41) and the answer was that emmet is bundled with the vue language tools already so another emmet lsp shouldn't be necessary.

I tried to use emmet from the bundled tools in the same way - by removing the "emmet-lsp" from the vue language config

 ```toml
[[language]]
name = "vue"
language-servers = [ "vue-language-server", "emmet-lsp"]
```

however when i test it in the `<template>` in an sfc by adding `SelfClosing/` and hitting tab, nothing happens.

I have also tried using the autocomplete (ctrl-x in INSERT mode) and the "code action" in NORMAL mode (space-a) but neither had any affect

Do you have any insight on how to use the tool please?

thanks
;oB

**Comment by daisnkzy** [comment]
same issue, are you solve it

**Reply by daisnkzy** [reply]
<img width="944" alt="截屏2024-08-11 12 00 29" src="https://github.com/user-attachments/assets/0b02448d-6637-4c2d-8f63-0c93a989dc2b">
set this, not perfect but working
